### PR TITLE
fix hybrid test for multi platforms

### PR DESCRIPTION
**Pull request by huminzhi**

### CODE REVIEW

**Comment by wmathurin** [comment]
Maybe we could change line 92-93 instead from
```javascript
if (testingIOS) updatePluginRepo(tmpDir, OS.ios, pluginRepoDir, sdkBranch);
if (testingAndroid) updatePluginRepo(tmpDir, OS.android, pluginRepoDir, sdkBranch);
```
To
```javascript
if (testingIOS && testingAndroid) updatePluginRepo(tmpDir, 'all', pluginRepoDir, sdkBranch);
if (testingIOS && !testingAndroid) updatePluginRepo(tmpDir, OS.ios, pluginRepoDir, sdkBranch);
if (testingAndroid && !testingIOS) updatePluginRepo(tmpDir, OS.android, pluginRepoDir, sdkBranch);
}
```

**Comment by wmathurin** [comment]
You could even make 'all' a value in the enum OS (line 20)
```javascript
var OS = {
    ios: 'ios',
    android: 'android'
    all: 'all'
};
```
Then you would pass OS.all instead of 'all' in line 92 suggested above.

**Comment by huminzhi** [comment]
@wmathurin hit an error as below if just add "all" as OS enum..  so I will use your first suggestion
path.js:7
    throw new TypeError('Path must be a string. Received ' + inspect(path));
    ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:1211:7)
    at createCompileApp (/Users/mhu/dev/SalesforceMobileSDK/SalesforceMobileSDK-Package/test/test_force.js:175:26)

**Comment by wmathurin** [comment]
I forgot the comma after 'android', did you have it?

**Comment by huminzhi** [comment]
@wmathurin yes, I have it.. it failed in var forcePath = path.join(tmpDir, 'node_modules', '.bin', FORCE_CLI[os]);

**Comment by wmathurin** [comment]
You passed --os=all from the command line?
There is more work if we want to allow 'all' as a supported value from the command line.
So maybe just pass 'all' in the updatePluginRepo call above. We can add support for all in 5.1 or later (or never).

File: test/test_force.js [hunk 1]
```diff
@@ -30,7 +30,8 @@ var APP_TYPE = {
 
 var FORCE_CLI = {
     ios: 'forceios',
-    android: 'forcedroid'
+    android: 'forcedroid',
+    all: 'all'
```
Comment (wmathurin) [inline test/test_force.js]: I would revert that change.

File: test/test_force.js [hunk 2]
```diff
@@ -87,8 +88,9 @@ function main(args) {
         if (pluginRepoUri.indexOf('//') >= 0) {
             // Actual uri - clone repo - run tools/update.sh
             var pluginRepoDir = utils.cloneRepo(tmpDir, pluginRepoUri);
-            if (testingIOS) updatePluginRepo(tmpDir, OS.ios, pluginRepoDir, sdkBranch);
-            if (testingAndroid) updatePluginRepo(tmpDir, OS.android, pluginRepoDir, sdkBranch);
+            if (testingIOS && testingAndroid) updatePluginRepo(tmpDir, OS.all, pluginRepoDir, sdkBranch);
```
Comment (wmathurin) [inline test/test_force.js]: OS.all is not defined, so I would just use 'all'

File: test/test_force.js [hunk 3]
```diff
@@ -156,7 +158,7 @@ function createDeployForcePackage(tmpDir, os) {
 //
 function updatePluginRepo(tmpDir, os, pluginRepoDir, sdkBranch) {
     utils.logInfo('Updating cordova plugin at ' + sdkBranch);
-    utils.runProcessThrowError(path.join('tools', 'update.sh') + ' -b ' + sdkBranch + ' -o ' + os, pluginRepoDir);
+    utils.runProcessThrowError(path.join('tools', 'update.sh') + ' -b ' + sdkBranch + ' -o all', pluginRepoDir);
```
Comment (wmathurin) [inline test/test_force.js]: I would revert that change

Comment (wmathurin) [inline test/test_force.js]: Since all will be passed when the user has specified --os=ios,android

**Pull request closed without merging**